sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslUdfUdafTest.java

```diff
@@ -101,6 +104,21 @@ public void testJodaTimeUdfUdaf() throws Exception {
     pipeline.run().waitUntilFinish();
   }
 
+  @Test
+  public void testListUdf() throws Exception {
```

Comment:
We should separate this into two modular test cases, one with a UDF that takes an array as an argument (e.g. the length function), and one with a UDF that returns an array (e.g. array constructor, like your previous test).

Reply:
Done.

Reply:
The tests LGTM, but we should separate the two blocks into separate `@Test` methods. That way we can easily tell if one case fails but the other succeeds. And it keeps the tests smaller and easier to read.

Reply:
I'm fine with doing this in a follow up PR, since the other tests in this class follow the same pattern.